x/language.rb

```diff
@@ -0,0 +1,18 @@
+module X
+  class Language
+    def self.of(id)
+      tracks[id.to_s]
+    end
+
+    def self.tracks
+      @@tracks ||= fetch_tracks.inject({}) do |hash, track|
+        hash[track.id] = track.language
+        hash
+      end
+    end
+
+    def self.fetch_tracks
+      @@fetched_tracks ||= Track.all
+    end
```

Comment:
There's an interesting edge case here that I'd like to handle correctly.

If we launch a new track, we redeploy the x-api app, and restart the application. We don't necessarily restart the exercism.io application.

If someone submits a solution on the new track, then the language will come up nil here, even though we do have it.

In this case I'd like to make the call again, and replace the cached value. Does that make sense?